### ISSUE TITLE
Windows 10 Mobile Error: DeviceInformation.FindAllAsync(WiFiAdapter.GetDeviceSelector()) crashs

### ISSUE THREAD

**Issue by sdechi** [Microsoft/Windows-universal-samples#170]
On the actual Windows Mobile build on my Lumia 930, the Method 
DeviceInformation.FindAllAsync(WiFiAdapter.GetDeviceSelector()); is crashing with an Exception saying:"The parameter is incorrect"

It works on Windows 10 desktop version

UPDATE: It works on the actual Mobile build of the Lumia 830... seems to be a problem of the 930?

Greetings,
Sascha

**Comment by HelgeKeck** [comment]
i have the same Problem, do you already have an solution?

**Comment by Sohra** [comment]
I also can't get this sample to work, I have a Lumia 930, and a Surface Pro 3.

I can't even begin to describe exactly what happens, because I regularly experience different behaviour with each attempt.  Sometimes I get prompted on the advertiser end to connect, but regardless of which, it will always fail to connect, and usually due to a timeout.  Never once have I ever had a session appear in the Connected Devices window.

Right now using the Lumia as connector, the advertiser (SP3) asks to accept the connection, and then both have the status of "Connecting to <other device name>" then they both ultimately timeout (first the advertiser, then the connector much later).

If I reverse roles and connect from the SP3, the Lumia doesn't ask for permission to connect and both apps will display "Devices connected on L2, listening on IP Address: 192.168.173.1 Port: 50001", but then both also timeout, and eventually the status changes to Disconnected.

I've seen a bunch of other behaviours too, but I'd be going from memory to remember the exact error codes and messages.  Numerous times (after different combinations of behaviour I've seen) "a method was called at an unexpected time HRESULT: 0x8000000E...), whether that was attempting to abort advertising or some other operation.  I've seen "Only one usage of each socket address (protocol/network address/port) is normally permitted."

**Comment by GauravC12** [comment]
A few same issues being experienced as mentioned above. Unfortunately couldn't figure out exact problem. Have tried with a few bunch of devices in different combination. No success!

**Comment by Sohra** [comment]
Tested again after installing KB3097617 this morning, and pulling down the October 2015 Update 2 commit, still can't establish anything more than "Devices connected on L2" and "waiting for client to connect."  I have not however, tested using the new 10549 Mobile build (waiting for one I can upgrade to from 10536.1004).

**Comment by Sohra** [comment]
Tested again after upgrading 930 to build 10549.  When functioning as the advertiser and another device attempts to connect to it (10240), it throws "FromIdAsync task threw an exception: System.Exception: Element not found (Exception from HRESULT: 0x80070490)"

When reversing the roles, the 930 asks me to enter a PIN to pair, after entering a 'Q' it says "Connection failed" and on the other device always throws the same exception quoted above.  Tried again entering a number and after disabling the textbox, it sits there waiting - meanwhile the watching end again, has the 80070490 error.

**Comment by sevenate** [comment]
Just adding my 0.2c - experiencing same error ("The parameter is incorrect") on **Lumia 635** with OS 10.0.**10549**.4. And it works fine on Windows 10 Pro Build 10565.

**Comment by sevenate** [comment]
Ok, calling [WiFiAdapter.FindAllAdaptersAsync()](https://msdn.microsoft.com/en-us/library/windows/apps/windows.devices.wifi.wifiadapter.findalladaptersasync) instead did the trick for me.

**Comment by Sohra** [comment]
After resolving a problem not being able to deploy the app to my mobile for quite some time ([information here](http://answers.microsoft.com/en-us/insider/forum/insider_wintp-insider_perf/lumia-640-lte-at-10572-no-longer-found-by-visual/5c60c8d0-a07b-4342-9f98-d292299b430d?auth=1) worked for me), I tested again running 10586.11 on both SP3 and Lumia 930.

With the phone operating as advertiser and the SP3 attempting to connect to it, it first failed with "Connect operation threw an exception: A device attached to the system is not functioning. (Exception from HRESULT: 0x8007001F)"

I reversed the roles, and was able to establish a session, and successfully send a message from each device to the other.

Reversing back to the original test, now that the two devices had been paired successfully, having the phone function as advertiser I was able to again establish a connection, and send a message from each device to the other.
